File: androidApp/src/main/kotlin/com/example/app/android/screens/MyPostsScreen.kt
package com.example.app.android.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.outlined.Article
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.res.stringResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.app.android.R
import com.example.app.android.components.FeedPostItem
import com.example.app.android.network.ApiClient
import com.example.app.android.network.models.FeedPost
import com.example.app.android.theme.AppTheme

@Composable
fun MyPostsScreen(modifier: Modifier = Modifier) {
    val colors = AppTheme.colors
    var posts by remember { mutableStateOf<List<FeedPost>>(emptyList()) }
    var isLoading by remember { mutableStateOf(true) }

    LaunchedEffect(Unit) {
        isLoading = true
        ApiClient.getMyPosts().onSuccess { posts = it }
        isLoading = false
    }

    Column(modifier = modifier.fillMaxSize()) {
        // Header
        Text(
            text = stringResource(R.string.my_posts_title),
            fontSize = 20.sp,
            fontWeight = FontWeight.SemiBold,
            color = colors.textPrimary,
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp)
                .padding(top = 60.dp, bottom = 16.dp)
        )

        when {
            isLoading -> {
                Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    CircularProgressIndicator(
                        color = colors.accent,
                        strokeWidth = 2.dp,
                        modifier = Modifier.size(24.dp)
                    )
                }
            }
            posts.isEmpty() -> {
                Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
                    Column(horizontalAlignment = Alignment.CenterHorizontally) {
                        Icon(
                            Icons.Outlined.Article,
                            contentDescription = null,
                            tint = colors.textSecondary.copy(alpha = 0.5f),
                            modifier = Modifier.size(32.dp)
                        )
                        Spacer(Modifier.height(4.dp))
                        Text(
                            text = stringResource(R.string.my_posts_empty),
                            fontSize = 14.sp,
                            color = colors.textSecondary
                        )
                    }
                }
            }
            else -> {
                LazyColumn(modifier = Modifier.fillMaxSize()) {
                    items(posts, key = { it.id }) { post ->
                        FeedPostItem(post = post)
                    }
                }
            }
        }
    }
}

File: iosApp/iosApp/Screens/MyPostsScreen.swift
import SwiftUI

struct MyPostsScreen: View {
    @Environment(\.appColors) private var colors
    @EnvironmentObject var lang: LanguageManager

    @State private var posts: [ApiFeedPost] = []
    @State private var isLoading = true

    var body: some View {
        VStack(spacing: 0) {
            // Header
            Text(lang.s("my_posts_title"))
                .font(.system(size: 20, weight: .semibold))
                .foregroundColor(colors.textPrimary)
                .frame(maxWidth: .infinity, alignment: .leading)
                .padding(.horizontal, 20)
                .padding(.top, 60)
                .padding(.bottom, 16)

            if isLoading {
                Spacer()
                ProgressView()
                    .tint(colors.accent)
                Spacer()
            } else if posts.isEmpty {
                Spacer()
                VStack(spacing: 4) {
                    Image(systemName: "doc.text")
                        .font(.system(size: 32))
                        .foregroundColor(colors.textSecondary.opacity(0.5))
                    Text(lang.s("my_posts_empty"))
                        .font(.system(size: 14))
                        .foregroundColor(colors.textSecondary)
                }
                Spacer()
            } else {
                ScrollView {
                    LazyVStack(spacing: 0) {
                        ForEach(posts) { post in
                            FeedPostItem(post: post)
                        }
                    }
                }
            }
        }
        .frame(maxWidth: .infinity, maxHeight: .infinity)
        .task {
            await loadPosts()
        }
    }

    private func loadPosts() async {
        do {
            posts = try await ApiClient.shared.getMyPosts()
        } catch {
            #if DEBUG
            print("Failed to load my posts: \(error)")
            #endif
        }
        isLoading = false
    }
}
